src/main/java/Dictionary.java

```diff
@@ -0,0 +1,89 @@
+import com.google.common.base.Charsets;
+import com.google.common.base.Stopwatch;
+import com.google.common.base.Supplier;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Sets;
+import com.google.common.io.Resources;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+
+public class Dictionary {
+
+    private final Logger logger= LogManager.getLogger("Dictionary");
+    private static final int LENGTH = 10;
+    private Multimap<Character,String> sortedDictByLetter;
+    private Multimap<Integer,String> sortedDictBySize;
+    private HashSet<String> input;
+
+    private List<String> words;
+
+    Dictionary()  {
+        sortedDictByLetter = HashMultimap.create();
+        sortedDictBySize = HashMultimap.create();
+    }
+
+    public void initialize() throws IOException {
```

Comment:
Must not throw IOException but handle it locally and log it. It is not something that a consumer of the method should take care of.